src/panels/config/automation/ha-automation-editor.ts

```diff
@@ -580,6 +614,137 @@ export class HaAutomationEditor extends PreventUnsavedMixin(
     }
   }
 
+  public connectedCallback() {
+    super.connectedCallback();
+    window.addEventListener("paste", this._handlePaste);
+  }
+
+  public disconnectedCallback() {
+    window.removeEventListener("paste", this._handlePaste);
+    super.disconnectedCallback();
+  }
+
+  private _handlePaste = async (ev: ClipboardEvent) => {
+    if (!canOverrideAlphanumericInput(ev.composedPath())) {
+      return;
+    }
+
+    const paste = ev.clipboardData?.getData("text");
+    if (!paste) {
+      return;
+    }
+
+    const loaded: any = load(paste);
+    if (loaded) {
+      let normalized: AutomationConfig | undefined;
+
+      try {
+        normalized = normalizeAutomationConfig(loaded);
+      } catch (_err: any) {
+        return;
+      }
+
+      try {
+        assert(normalized, automationConfigStruct);
+      } catch (_err: any) {
+        showToast(this, {
+          message: this.hass.localize(
+            "ui.panel.config.automation.editor.paste_invalid_config"
+          ),
+          duration: 4000,
+          dismissable: true,
+        });
+        return;
+      }
+
+      if (normalized) {
+        ev.preventDefault();
+
+        if (this._dirty) {
+          const result = await new Promise<boolean>((resolve) => {
+            showPasteReplaceDialog(this, {
+              domain: "automation",
+              pastedConfig: normalized,
+              onClose: () => resolve(false),
+              onAppend: () => {
+                this._appendToExistingConfig(normalized);
+                resolve(false);
+              },
+              onReplace: () => resolve(true),
+            });
+          });
+
+          if (!result) {
+            return;
+          }
+        }
+
+        // replace the config completely
+        this._replaceExistingConfig(normalized);
+      }
+    }
+  };
+
+  private _appendToExistingConfig(config: AutomationConfig) {
+    // make a copy otherwise we will reference the original config
+    this._previousConfig = { ...this._config } as AutomationConfig;
+    this._pastedConfig = config;
+
+    if (!this._config) {
+      return;
+    }
+
+    if (Array.isArray(this._config?.triggers)) {
+      this._config.triggers = this._config.triggers.concat(
+        config.triggers || []
+      );
+    }
+
+    if (Array.isArray(this._config?.conditions)) {
+      this._config.conditions = this._config.conditions.concat(
+        config.conditions || []
+      );
+    }
+
+    if (Array.isArray(this._config?.actions)) {
+      this._config.actions = this._config.actions.concat(config.actions || []);
+    }
```

Comment:
```suggestion
    if ("triggers" in config) {
      this._config.triggers = ensureArray(this._config.triggers || []).concat(
        ensureArray(config.triggers)
      );
    }

    if ("conditions" in config) {
      this._config.conditions = ensureArray(
        this._config.conditions || []
      ).concat(ensureArray(config.conditions));
    }

    if ("actions" in config) {
      this._config.actions = ensureArray(this._config.actions || []).concat(
        ensureArray(config.actions)
      );
    }

    this._config = { ...this._config };
```

The `triggers`, `conditions`, `actions` keys don't have to be present in the current automation, so we should look at the pasted config, not the current.